dqnroute/simpy_router.py

```diff
@@ -0,0 +1,60 @@
+import simpy
+import random
+
+class SimPyAbstractRouter(object):
+    def __init__(self, env, id):
+        self.env = env
+        self.id = id
+
+    def setNeighbours(self, neighbours):
+        self.neighbours = neighbours
+
+    def __sendSimulation(self, pkg, dest):
+        yield self.env.timeout(random.randint(5, 10))
+        # manual trigger getter on destination router
+        dest.getPackage(self.id, pkg)
+        
+    def __sendPkg(self, pkg, dest):
+        print("Sending packet: {} -> {} at time {}...".format(self.id, dest.id, self.env.now))
+
+        if self.id == dest.id:
+            print("Got source router {} as the destination router".format(self.id))
+            # actual process of pkg & empty yielding
+            yield self.env.timeout(0)
+        else:
+            # actual sending process & simulation
+            self.env.process(self.__sendSimulation(pkg, dest))
+        #print("Sent from router {} at time {}".format(self.id, self.env.now))
```

Comment:
You have quite a few commented prints which I assume you are using for debugging. Maybe it would make sense to add some simple logging framework and use it instead of manually add/remove print lines every time you want to debug something.